lane.py

```diff
@@ -725,14 +725,29 @@ def works(self, _db, facets=None, pagination=None, include_quality_tier=False):
         # apply_filters() requires that the query include a join
         # against LicensePool. If nothing else, the `facets` may
         # restrict the query to currently available items.
-        qu = qu.join(LicensePool, LicensePool.id==mw.license_pool_id)
-        qu = qu.options(contains_eager(mw.license_pool))
+        qu = qu.join(mw.license_pool)
         if self.collection_ids is not None:
             qu = qu.filter(
                 LicensePool.collection_id.in_(self.collection_ids)
             )
-
         qu = self.apply_filters(_db, qu, facets, pagination)
+        if qu:
+            qu = qu.options(
+                contains_eager(mw.license_pool),
+                # TODO: Strictly speaking, these joinedload calls are 
+                # only needed by the circulation manager. This code could
+                # be moved to circulation and everyone else who uses this
+                # would be a little faster. (But right now there is no one
+                # else who uses this.)
+
+                # These speed up the process of generating acquisition links.
+                joinedload("license_pool", "delivery_mechanisms"),
```

Comment:
This line is why I had to make LicensePool.delivery_mechanisms into a proper relationship -- otherwise there was no way to tell SQLAlchemy to load in the LicensePoolDeliveryMechanism objects or any of that object's children.